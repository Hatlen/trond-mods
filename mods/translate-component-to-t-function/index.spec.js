import pluginTester from "babel-plugin-tester";
import babelI18nPlugin from "./";

pluginTester({
  plugin: babelI18nPlugin,
  pluginOptions: {
    allTranslations: {
      en: {
        key: "value",
        key_with_pluralization: {
          one: "one value",
          many: "many value"
        },
        key_with_html: "<em>Welcome</em>",
        key_with_pluralization_and_html: {
          one: "no html",
          many: "<div>MANY!</div>"
        }
      },
      sv: {
        key: "värde",
        key_with_pluralization: {
          one: "1 värde"
        },
        key_with_html: "<em>Welcome</em>",
        key_with_pluralization_and_html: {
          one: "no html",
          many: "<div>MANY!</div>"
        }
      }
    }
  },
  babelOptions: {
    filename: ".babelrc",
    babelrc: true
  },
  snapshot: true,
  tests: [
    `
const C = (props) => (
  <div>
    <Translate
      count={props.totalVisitors}
      totalViews={numeral(props.totalVisitors).format("0,0")}
      i18nKey="key"
    />
  </div>
);

export default C;
        `.trim(),
    `
const C = () => (
  <div>
    <Translate
      count={props.totalVisitors}
      totalViews={numeral(props.totalVisitors).format("0,0")}
      i18nKey="key_with_pluralization"
    />
  </div>
);

export default C;
        `.trim(),
    `
// Simplified file without import or export
const C = () => (
  <div>
    <Translate
      count={props.totalVisitors}
      totalViews={numeral(props.totalVisitors).format("0,0")}
      i18nKey="key_with_html"
    />
  </div>
);

export default C;
        `.trim(),
    `
// Simplified file without import or export
const C = () => (
  <div>
    <Translate
      count={props.totalVisitors}
      totalViews={numeral(props.totalVisitors).format("0,0")}
      i18nKey="key_with_pluralization_and_html"
    />
  </div>
);

export default C;
        `.trim()
  ]
});
