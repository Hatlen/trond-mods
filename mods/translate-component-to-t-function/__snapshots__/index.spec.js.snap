// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Translate-to-withNamespaces-t 1. Translate-to-withNamespaces-t: 1. Translate-to-withNamespaces-t 1`] = `
"
import Translate from 'components/Translate';

const C = (props) => (
  <div>
    <Translate
      count={props.totalVisitors}
      totalViews={numeral(props.totalVisitors).format(\\"0,0\\")}
      i18nKey=\\"key\\"
    />
  </div>
);

export default C;

      ↓ ↓ ↓ ↓ ↓ ↓

\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.default = void 0;

var _reactI18next = require(\\"react-i18next\\");

const C = ({
  t,
  ...props
}) => React.createElement(\\"div\\", null, t(\\"key\\", {
  count: props.totalVisitors,
  totalViews: numeral(props.totalVisitors).format(\\"0,0\\")
}));

var _default = (0, _reactI18next.withNamespaces)(C);

exports.default = _default;
"
`;

exports[`Translate-to-withNamespaces-t 2. Translate-to-withNamespaces-t: 2. Translate-to-withNamespaces-t 1`] = `
"
import Translate from 'components/Translate';

const C = ({ t }) => (
  <div>
    <Translate
      count={props.totalVisitors}
      totalViews={numeral(props.totalVisitors).format(\\"0,0\\")}
      i18nKey=\\"key_with_pluralization\\"
    />
  </div>
);

export default C;

      ↓ ↓ ↓ ↓ ↓ ↓

\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.default = void 0;

var _reactI18next = require(\\"react-i18next\\");

const C = ({
  t
}) => React.createElement(\\"div\\", null, t(\\"key_with_pluralization\\", {
  count: props.totalVisitors,
  totalViews: numeral(props.totalVisitors).format(\\"0,0\\")
}));

var _default = (0, _reactI18next.withNamespaces)(C);

exports.default = _default;
"
`;

exports[`Translate-to-withNamespaces-t 3. Translate-to-withNamespaces-t: 3. Translate-to-withNamespaces-t 1`] = `
"
import Translate from 'components/Translate';

const C = ({ totalVisitors }) => (
  <div>
    <Translate
      count={totalVisitors}
      totalViews={numeral(totalVisitors).format(\\"0,0\\")}
      i18nKey=\\"key_with_html\\"
    />
  </div>
);

export default C;

      ↓ ↓ ↓ ↓ ↓ ↓

\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.default = void 0;

var _reactI18next = require(\\"react-i18next\\");

const C = ({
  t,
  totalVisitors
}) => React.createElement(\\"div\\", null, safeT(t(\\"key_with_html\\", {
  count: totalVisitors,
  totalViews: numeral(totalVisitors).format(\\"0,0\\")
})));

var _default = (0, _reactI18next.withNamespaces)(C);

exports.default = _default;
"
`;

exports[`Translate-to-withNamespaces-t 4. Translate-to-withNamespaces-t: 4. Translate-to-withNamespaces-t 1`] = `
"
import Translate from 'components/Translate';

const C = () => (
  <div>
    <Translate
      count={props.totalVisitors}
      totalViews={numeral(props.totalVisitors).format(\\"0,0\\")}
      i18nKey=\\"key_with_pluralization_and_html\\"
    />
    {something ? (<Translate i18nKey=\\"key\\" />) : (<Translate i18nKey=\\"key\\" />)}
  </div>
);

export default C;

      ↓ ↓ ↓ ↓ ↓ ↓

\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.default = void 0;

var _reactI18next = require(\\"react-i18next\\");

const C = ({
  t
}) => React.createElement(\\"div\\", null, safeT(t(\\"key_with_pluralization_and_html\\", {
  count: props.totalVisitors,
  totalViews: numeral(props.totalVisitors).format(\\"0,0\\")
})), something ? t(\\"key\\") : t(\\"key\\"));

var _default = (0, _reactI18next.withNamespaces)(C);

exports.default = _default;
"
`;

exports[`Translate-to-withNamespaces-t 5. Translate-to-withNamespaces-t: 5. Translate-to-withNamespaces-t 1`] = `
"
import Translate from 'components/Translate';

const C = (props) => (
  <div>
    <Translate
      count={props.totalVisitors}
      totalViews={numeral(props.totalVisitors).format(\\"0,0\\")}
      i18nKey=\\"key_with_pluralization_and_html\\"
    />
    {props.t('key')}
  </div>
);

export default C;

      ↓ ↓ ↓ ↓ ↓ ↓

\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.default = void 0;

var _reactI18next = require(\\"react-i18next\\");

const C = ({
  t,
  ...props
}) => React.createElement(\\"div\\", null, safeT(t(\\"key_with_pluralization_and_html\\", {
  count: props.totalVisitors,
  totalViews: numeral(props.totalVisitors).format(\\"0,0\\")
})), t('key'));

var _default = (0, _reactI18next.withNamespaces)(C);

exports.default = _default;
"
`;

exports[`Translate-to-withNamespaces-t 6. Translate-to-withNamespaces-t: 6. Translate-to-withNamespaces-t 1`] = `
"
import { Component } from 'react';
import Translate from 'components/Translate';

class C extends Component {
  render() {
    const { totalVisitors } = this.props;

    return (
      <div>
        <Translate
          count={totalVisitors}
          totalViews={numeral(this.props.totalVisitors).format(\\"0,0\\")}
          i18nKey=\\"key_with_pluralization_and_html\\"
        />
        {this.props.t('key')}
      </div>
    );
  }
}

export default C;

      ↓ ↓ ↓ ↓ ↓ ↓

\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
exports.default = void 0;

var _reactI18next = require(\\"react-i18next\\");

var _react = require(\\"react\\");

class C extends _react.Component {
  render() {
    const {
      t,
      totalVisitors
    } = this.props;
    return React.createElement(\\"div\\", null, safeT(t(\\"key_with_pluralization_and_html\\", {
      count: totalVisitors,
      totalViews: numeral(this.props.totalVisitors).format(\\"0,0\\")
    })), t('key'));
  }

}

var _default = (0, _reactI18next.withNamespaces)(C);

exports.default = _default;
"
`;
